packages/ui/src/views/repo/pull-request/pull-request-list-page.tsx

```diff
@@ -137,50 +159,127 @@ const PullRequestList: FC<PullRequestPageProps> = ({
+                <ListActions.Left>
+                  <SearchBox.Root
+                    width="full"
+                    className="max-w-96"
+                    value={searchInput || ''}
+                    handleChange={handleInputChange}
+                    placeholder={t('views:repos.search', 'Search')}
+                  />
+                </ListActions.Left>
+                <ListActions.Right>
+                  <PRListFilterHandler.Dropdown>
+                    {(addFilter, availableFilters, resetFilters) => (
+                      <FilterSelect
+                        options={PR_FILTER_OPTIONS.filter(option =>
+                          availableFilters.includes(option.value as PRListFiltersKeys)
+                        )}
+                        onChange={option => {
+                          addFilter(option.value as PRListFiltersKeys)
+                          setOpenedFilter(option.value as PRListFiltersKeys)
+                        }}
+                        onReset={resetFilters}
+                        inputPlaceholder={t('component:filter.inputPlaceholder', 'Filter by...')}
+                        buttonLabel={t('component:filter.buttonLabel', 'Reset filters')}
+                        displayLabel={
+                          <FilterSelectLabel
+                            selectedFilters={PR_FILTER_OPTIONS.length - availableFilters.length}
+                            displayLabel={t('component:filter.defaultLabel', 'Filter')}
+                          />
+                        }
+                      />
+                    )}
+                  </PRListFilterHandler.Dropdown>
+                  <FilterTrigger
+                    type="sort"
+                    activeFilters={filterHandlers.activeSorts}
+                    onChange={filterHandlers.handleSortChange}
+                    onReset={filterHandlers.handleResetSorts}
+                    searchQueries={filterHandlers.searchQueries}
+                    onSearchChange={filterHandlers.handleSearchChange}
+                    options={SORT_OPTIONS}
+                    t={t}
+                  />
+                  <Button variant="default" asChild>
+                    <Link to={`${spaceId ? `/${spaceId}` : ''}/repos/${repoId}/pulls/compare/`}>New pull request</Link>
+                  </Button>
+                </ListActions.Right>
+              </ListActions.Root>
+              <ListControlBar<PRListFilters>
+                renderSelectedFilters={filterFieldRenderer => (
+                  <PRListFilterHandler.Content className={'flex items-center gap-x-2'}>
+                    {PR_FILTER_OPTIONS.map(filterOption => (
+                      <PRListFilterHandler.Component
+                        parser={
+                          'parser' in filterOption
+                            ? // TODO Need to address the type issue here
+                              (filterOption.parser as unknown as Parser<PRListFilters[PRListFiltersKeys]>)
```

Comment:
let's discuss what's the issue here. this looks dangerous

Reply:
Seems like this is needed for now. Will find a proper fix with @harish-viswa as a follow up.